contracts/contracts/coordination/SigningCoordinator.sol

```diff
@@ -0,0 +1,377 @@
+        SigningCohort storage signingCohort = signingCohorts[cohortId];
+        require(signingCohort.chains.length > 0, "Initial chain not yet deployed");
+        for (uint256 i = 0; i < signingCohort.chains.length; i++) {
+            require(signingCohort.chains[i] != chainId, "Already deployed");
+        }
+        _deploySigningMultisig(chainId, cohortId);
+        signingCohort.chains.push(chainId);
+    }
+
+    function _deploySigningMultisig(uint256 _chainId, uint32 _cohortId) internal {
+        SigningCohort storage signingCohort = signingCohorts[_cohortId];
+        require(_isCohortActive(signingCohort), "Cohort not active");
+
+        address[] memory _signers = new address[](signingCohort.numSigners);
+        for (uint256 i = 0; i < signingCohort.signers.length; i++) {
+            // ursula operator address does signing; not staking provider
+            _signers[i] = signingCohort.signers[i].operator;
+        }
+
+        signingCoordinatorDispatcher.dispatch(
+            _chainId,
+            abi.encodeWithSelector(
+                ISigningCoordinatorChild.deployCohortMultiSig.selector,
+                _cohortId,
+                _signers,
+                signingCohort.threshold
+            )
+        );
+        emit SigningCohortDeployed(_cohortId, _chainId);
+    }
+
+    function getSigningCohortState(uint32 cohortId) public view returns (SigningCohortState) {
+        return _getSigningCohortState(signingCohorts[cohortId]);
+    }
+
+    function _getSigningCohortState(
+        SigningCohort storage _signingCohort
+    ) internal view returns (SigningCohortState) {
+        uint32 t0 = _signingCohort.initTimestamp;
+        uint32 deadline = t0 + timeout;
+        if (t0 == 0) {
+            return SigningCohortState.NON_INITIATED;
+        } else if (_signingCohort.totalSignatures == _signingCohort.numSigners) {
+            // Cohort formation was successful
+            if (block.timestamp <= _signingCohort.endTimestamp) {
+                return SigningCohortState.ACTIVE;
+            }
+            return SigningCohortState.EXPIRED;
+        } else if (block.timestamp > deadline) {
+            // DKG failed due to timeout
+            return SigningCohortState.TIMEOUT;
+        } else if (_signingCohort.totalSignatures < _signingCohort.numSigners) {
+            return SigningCohortState.AWAITING_SIGNATURES;
+        } else {
+            /**
+             * It shouldn't be possible to reach this state:
+             *   - No public key
+             *   - All transcripts and all aggregations
+             *   - Still within the deadline
+             */
+            revert("Ambiguous ritual state");
+        }
+    }
+
+    function getSigningCoordinatorChild(uint256 chainId) external view returns (address) {
+        address child = signingCoordinatorDispatcher.getSigningCoordinatorChild(chainId);
+        require(child != address(0), "No child application found");
```

Comment:
considering that this is a view maybe no need to revert tx? basically outside caller will decide what to do with that if there is no child?

Reply:
Good point - ✔️